telegram/ext/dispatcher.py

```diff
@@ -86,16 +94,30 @@ class Dispatcher(object):
     __singleton = None
     logger = logging.getLogger(__name__)
 
-    def __init__(self, bot, update_queue, workers=4, exception_event=None, job_queue=None):
+    def __init__(self, bot, update_queue, workers=4, exception_event=None, job_queue=None,
+                 persistence=None):
         self.bot = bot
         self.update_queue = update_queue
-        self.job_queue = job_queue
         self.workers = workers
-
         self.user_data = defaultdict(dict)
-        """:obj:`dict`: A dictionary handlers can use to store data for the user."""
         self.chat_data = defaultdict(dict)
-        """:obj:`dict`: A dictionary handlers can use to store data for the chat."""
+        if persistence:
+            if not isinstance(persistence, BasePersistence):
+                self.logger.warning("persistence should be based on telegram.ext.BasePersistence")
```

Comment:
Imo this should raise an exception

Reply:
I thought about that. But if someone really wants to create their own version without basing it on Basepersistence they should be allowed

Reply:
I equate it to how we handle handlers in dispatche https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/ext/dispatcher.py#L329 
Also BaseFilter iirc

Reply:
ok seems legit